pkg/apis/provisioning/v1alpha5/constraints.go

```diff
@@ -58,13 +58,17 @@ func (c *Constraints) ToNode() *v1.Node {
 		labels[key] = value
 	}
 	for key := range c.Requirements.Keys() {
-		if !IsRestrictedNodeLabel(key) {
-			switch c.Requirements.Get(key).Type() {
-			case v1.NodeSelectorOpIn:
+		if IsRestrictedNodeLabel(key) {
+			continue
+		}
+		switch c.Requirements.Get(key).Type() {
```

Comment:
Thoughts on only applying this logic to custom labels? Well known labels should be determined by the cloud provider, because only it can know specifically which node it launched given the zone/instance type flexibility.

Reply:
The problem is that us creating node objects breaks that. If we create a node and don't put an OS label on it, we don't know that a pod with a node selector for OS can schedule to it. We then launch extra nodes.

Reply:
I meant here: https://github.com/aws/karpenter/blob/main/pkg/cloudprovider/aws/instance.go#L283. I *think* we can know OS from the AMI type. https://aws.amazon.com/premiumsupport/knowledge-center/ec2-linux-find-os-platform-or-version/

Reply:
Yeah, that's better, I'm going to delete this and create similar tests in the "no node creation/pod binding" branch